src/GafferScene/Instancer.cpp

```diff
@@ -2535,39 +2528,44 @@ void Instancer::PrototypeScope::setPrototype( const EngineData *engine, const Sc
 namespace
 {
 
-void addPrototypeToVectors(
+struct Prototype
+{
+	ConstObjectPtr object;
+	ConstCompoundObjectPtr attributes;
+	std::vector<M44f> transforms;
+};
+
+Prototype addPrototypeToVectors(
```

Comment:
Rather than makePrototype, I've just made this a constructor of the Prototype struct? Probably reasonable ... the one possibly naughty thing I've done is used `m_` prefixes on the members, since that seemed helpful to keep track ... but maybe that is supposed to imply privacy in our convention?

Reply:
Constructor seems ideal to me. You're right about the `m_` convention - seems like it might not be too much of a leap to actually make those private and add a public `Prototype::render( const M44f &instanceTransform, const CompoundObject instanceAttributes )` method?

Reply:
Dangit, after addressing everything else and messing around with a bit extra, I went to upload the PR at the end of the day ... and then realized I'd totally missed this comment. It does make sense, and I've now done it in a final fixup commit, though in a bit more of a hurry than it should have been. I think it's all looking pretty decent, interface seems mostly pretty clean. I started with a slightly simpler version where I passed the instanceAttributes as a CompoundObject - but that required an extra copy of the CompoundObject ... to measure any difference, I had to beef up the attributes perf test a bit, but I did end up seeing a 10% slowdown - which I was able to get back to parity with the previous version of the code by passing a functor that processes the attributes.